include/cpuinfo.h

```diff
@@ -18,15 +18,15 @@
 #define CPUINFO_ARCH_X86 1
 #endif
 
-#if defined(__x86_64__) || defined(__x86_64) || defined(_M_X64) || defined(_M_AMD64)
+#if defined(__x86_64__) || defined(__x86_64) || (defined(_M_X64) && !defined(_M_ARM64EC)) || (defined(_M_AMD64) && !defined(_M_ARM64EC))
```

Comment:
I checked https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170
MSVC architecture related defination is complex, suggest to add more comments.